runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiObjectSMREntry.java

```diff
@@ -101,20 +92,39 @@ public void serialize(ByteBuf b) {
                 .forEach(x -> {
                     b.writeLong(x.getKey().getMostSignificantBits());
                     b.writeLong(x.getKey().getLeastSignificantBits());
-                    Serializers.CORFU.serialize(x.getValue(), b);
+
+                    int lengthIndex = b.writerIndex();
+                    b.writeInt(0);
+                    b.writeInt(x.getValue().size());
+                    x.getValue().stream().forEach(smrEntry -> Serializers.CORFU.serialize(smrEntry, b));
+                    int length = b.writerIndex() - lengthIndex - 4;
+                    b.writerIndex(lengthIndex);
+                    b.writeInt(length);
+                    b.writerIndex(lengthIndex + length + 4);
```

Comment:
This can be simplified with 
```java

                    int payloadSizeIndex = b.writerIndex();
                    b.writeInt(0);
                    int payloadIndex = b.writerIndex();
                    b.writeInt(x.getValue().size());
                    x.getValue().stream().forEach(smrEntry -> Serializers.CORFU.serialize(smrEntry, b));
                    int length = b.writerIndex() - payloadIndex;
                    b.writerIndex(payloadSizeIndex);
                    b.writeInt(length);
                    b.writerIndex(payloadIndex + length);

```